invenio_communities/assets/semantic-ui/js/invenio_communities/community/communities-carousel/CommunitiesCarousel.js

```diff
@@ -0,0 +1,174 @@
+/*
+ * This file is part of Invenio.
+ * Copyright (C) 2016-2022 CERN.
+ *
+ * Invenio is free software; you can redistribute it and/or modify it
+ * under the terms of the MIT License; see LICENSE file for more details.
+ */
+
+import React, { Component } from "react";
+import PropTypes from "prop-types";
+import _truncate from "lodash/truncate";
+import { i18next } from "@translations/invenio_communities/i18next";
+import { http } from "../../api/config";
+import { withCancel } from "react-invenio-forms";
+import { Transition, Container, Grid, Header, Item, Icon } from "semantic-ui-react";
+import CarouselItem from "./CarouselItem";
+
+export default class CommunitiesCarousel extends Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      isLoading: false,
+      data: { hits: [] },
+      error: {},
+      activeIndex: 0,
+      animationDirection: 'left',
+      carouselTimer: null,
+    };
+  }
+
+  getDataIndex = (index) => {
+    const dataLength = this.state.data.hits.length;
+    if(index > dataLength - 1) return 0;
+    if(index < 0) return dataLength - 1;
+    return index;
+  }
+
+  runCarousel = async (newIndex) => {
+    const { activeIndex } = this.state;
+    let animationDirection = newIndex < activeIndex ? 'right' : 'left';
+    await this.setState({ animationDirection });
+    this.setState({ activeIndex: this.getDataIndex(newIndex) })
+  }
+
+  setCarouselTimer = () => {
+    const { data } = this.state;
+    const { intervalDelay } = this.props;
+    this.setState({
+      carouselTimer: setInterval(() => {
+        data.hits.length && this.runCarousel(this.state.activeIndex + 1);
+      }, intervalDelay)
+    })
+  }
+
+  // Pause carousel when user focuses elements in the container
+  stopCarousel = () => { clearInterval(this.state.carouselTimer) }
+  startCarousel = () => { this.setCarouselTimer() }
+
+  componentWillUnmount() {
+    this.cancellableFetch && this.cancellableFetch.cancel();
+    this.stopCarousel();
+  }
+
+  componentDidMount() {
+    this.fetchData();
+  }
+
+  fetchData = async () => {
+    this.setState({ isLoading: true });
+    this.cancellableFetch = withCancel(http.get(this.props.fetchUrl));
+
+    try {
+      const response = await this.cancellableFetch.promise;
+      this.setState({ data: response.data.hits, isLoading: false });
+      this.setCarouselTimer();
+
+    } catch (error) {
+      console.log(error);
```

Comment:
```suggestion
      console.error(error);
```
what about displaying an error message for the user ? otherwise if the data loading fails, the component will be in loading state forever, are we OK with that?